### PR TITLE
Add French language and file_date method

### PR DESCRIPTION
**Pull request by ArthurFDLR**
* Add adequate headers for a French N26 CSV statement

    ```python
    'fr': OrderedDict(
        (
            ('date', 'Date'),
            ('payee', 'Bénéficiaire'),
            ('account_number', 'Numéro de compte'),
            ('transaction_type', 'Type de transaction'),
            ('payment_reference', 'Référence de paiement'),
            ('category', 'Catégorie'),
            ('amount_eur', 'Montant (EUR)'),
            ('amount_foreign_currency', 'Montant (Devise étrangère)'),
            ('type_foreign_currency', 'Sélectionnez la devise étrangère'),
            ('exchange_rate', 'Taux de conversion'),
        )
    )
    ```
* Add file_date method to organize new statements with bean-file command

### CODE REVIEW

File: beancount_n26/__init__.py [hunk 1]
```diff
@@ -85,6 +99,20 @@ def name(self):
     def file_account(self, _):
         return self.account
 
+    def file_date(self, file_):
```
Comment (siddhantgoel) [inline beancount_n26/__init__.py:102]: Just to understand better, are you trying to return the most recent date from this function or the date in the last line?

File: beancount_n26/__init__.py [hunk 2]
```diff
@@ -85,6 +99,20 @@ def name(self):
...
+        date = None
+        
+        if not self.identify(file_):
+            return date
+        
+        with open(file_.name, encoding=self.file_encoding) as fd:
+            reader = csv.DictReader(
+                fd, delimiter=',', quoting=csv.QUOTE_MINIMAL, quotechar='"')
+            for line in reader:
+                finalLine = line
```
Comment (siddhantgoel) [inline beancount_n26/__init__.py:112]: `camel_case` please. :slightly_smiling_face:

**Review by siddhantgoel — CHANGES_REQUESTED** [review]
Thanks for the PR! The build is currently marked as red because of some linting errors. Could you please run `make lint` locally and fix the errors reported?

File: beancount_n26/__init__.py (hunk 1 above)
Comment (ArthurFDLR) [reply beancount_n26/__init__.py:102]: I was actually trying to return the most recent date which happened to be in the last line of my statements. I just update the method to clarify the behavior!

**Comment by siddhantgoel** [comment]
@ArthurFDLR thanks for the update! Could you please add 1-2 tests for this? Nothing super fancy but just something that asserts that out of 2-3 lines in the statement, the most recent date is returned. There are a few tests already that contain some test data. Feel free to reuse those.

**Comment by ArthurFDLR** [comment]
I just added a couple of tests. It is my first time using `pytest`, do not hesitate to tell me if I should modify something!

File: beancount_n26/__init__.py [hunk 3]
```diff
@@ -85,6 +99,27 @@ def name(self):
...
+            )
+
+            for line in reader:
+                date_tmp = datetime.strptime(
+                    line[self._translate('date')], '%Y-%m-%d'
+                )
+                if not date or date_tmp > date:
+                    date = date_tmp
+
+        return date.date()
```
Comment (siddhantgoel) [inline beancount_n26/__init__.py:121]: One last change request. For empty files, `date` would be `None` here so `date.date()` would not work. Could you replace this with `return date.date() if date else None`?

**Review by siddhantgoel — CHANGES_REQUESTED** [review]

File: beancount_n26/__init__.py (hunk 3 above)
Comment (ArthurFDLR) [reply beancount_n26/__init__.py]: Sure, it is sorted.

**Review by siddhantgoel — APPROVED** [review]
Nice :+1:

**Pull request merged**